x/concentrated-liquidity/incentives_test.go

```diff
@@ -3509,3 +3511,241 @@ func (s *KeeperTestSuite) TestPrepareBalancerPoolAsFullRange() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestClaimAndResetFullRangeBalancerPool() {
+	lockableDurations := s.App.PoolIncentivesKeeper.GetLockableDurations(s.Ctx)
+	longestLockableDuration := lockableDurations[len(lockableDurations)-1]
+	uptimeHelper := getExpectedUptimes()
+
+	tests := map[string]struct {
+		existingClLiquidity sdk.Coins
+		balancerPoolAssets  []balancer.PoolAsset
+		uptimeGrowth        []sdk.DecCoins
+
+		concentratedPoolDoesNotExist bool
+		balancerPoolDoesNotExist     bool
+		balSharesNotAddedToAccums    bool
+
+		expectedError error
+	}{
+		"happy path: valid CL and bal pool IDs": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.hundredTokensMultiDenom,
+		},
+		"valid pool IDs with no uptime growth": {
+			// 100 existing shares and 100 shares added from balancer
+			existingClLiquidity: defaultConcentratedAssets,
+			balancerPoolAssets:  defaultBalancerAssets,
+			uptimeGrowth:        uptimeHelper.emptyExpectedAccumValues,
+		},
+		"valid pool IDs with uneven uptime growth": {
```

Comment:
What makes it uneven?

Reply:
`varyingTokensMultiDenom` has different amounts for each denom